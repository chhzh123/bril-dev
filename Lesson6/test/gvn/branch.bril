@main(a0: int, b0: int, c0: int, d0: int, e0: int, f0: int) {
.B1:
  u0: int = add a0 b0;
  v0: int = add c0 d0;
  w0: int = add e0 f0;
  cond: bool = const true;
  br cond .B2 .B3;
.B2:
  x0: int = add c0 d0;
  y0: int = add c0 d0;
  jmp .B4;
.B3:
  u1: int = add a0 b0;
  x1: int = add e0 f0;
  y1: int = add e0 f0;
  jmp .B4;
.B4:
  u2: int = phi u0 u1 .B2 .B3;
  x2: int = phi x0 x1 .B2 .B3;
  y2: int = phi y0 y1 .B2 .B3;
  z0: int = add u2 y2;
  u3: int = add a0 b0;
}
