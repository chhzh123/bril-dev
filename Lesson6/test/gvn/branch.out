@main(a0: int, b0: int, c0: int, d0: int, e0: int, f0: int) {
.B1:
  u0: int = add a0 b0;
  v0: int = add c0 d0;
  w0: int = add e0 f0;
  cond: bool = const true;
  br cond .B2 .B3;
.B2:
  jmp .B4;
.B3:
  jmp .B4;
.B4:
  x2: int = phi v0 w0 .B2 .B3;
  z0: int = add u0 x2;
  ret;
}
